src/Http/Http/src/Features/ResponseBodyPipeFeature.cs

```diff
@@ -25,25 +27,21 @@ public PipeWriter Writer
         {
             get
             {
-                if (_userSetPipeWriter != null)
-                {
-                    return _userSetPipeWriter;
-                }
-
                 if (_internalPipeWriter == null ||
-                    !object.ReferenceEquals(_internalPipeWriter.InnerStream, _context.Response.Body))
+                    !ReferenceEquals(_streamInstanceWhenWrapped, _context.Response.Body))
                 {
-                    _internalPipeWriter = new StreamPipeWriter(_context.Response.Body);
-                    _context.Response.RegisterForDispose(_internalPipeWriter);
+                    _streamInstanceWhenWrapped = _context.Response.Body;
+                    _internalPipeWriter = PipeWriter.Create(_context.Response.Body);
```

Comment:
Do we want to specify the leaveOpen option here if and when it becomes available? Or do we want to dispose whatever someone set Response.Body to on their available. I'm leaning towards the former.

Reply:
We’d probably set leaveOpen to true and not have it be pluggable.